src/privatesend/privatesend-util.cpp

```diff
@@ -70,3 +82,250 @@ void CKeyHolderStorage::ReturnAll()
         LogPrintf("CKeyHolderStorage::%s -- %lld keys returned\n", __func__, tmp.size());
     }
 }
+
+CTransactionBuilderOutput::CTransactionBuilderOutput(CTransactionBuilder* pTxBuilderIn, CWallet* pwalletIn, CAmount nAmountIn) :
+    pTxBuilder(pTxBuilderIn),
+    key(pwalletIn),
+    nAmount(nAmountIn)
+{
+    assert(pTxBuilder);
+    CPubKey pubKey;
+    key.GetReservedKey(pubKey, false);
+    script = ::GetScriptForDestination(pubKey.GetID());
+}
+
+bool CTransactionBuilderOutput::UpdateAmount(const CAmount nNewAmount)
+{
+    LOCK(pTxBuilder->cs_outputs);
+
+    const CAmount nAmountDiff = nNewAmount - nAmount;
+
+    if (nNewAmount > 0 && nAmountDiff <= pTxBuilder->GetAmountLeft()) {
+        if (nAmountDiff != 0) {
+            nAmount = nNewAmount;
+        }
```

Comment:
I don't understand why this `if` statement is here

Reply:
Oh good catch :D thats a leftover from an earlier iteration of `UpdateAmount` where an amount update triggered some recalculations which would have been be avoided by that if, see 7814ff3bfd393757c4c014b92d9f90e9da3deda0